src/overlay016/battle_controller.c

```diff
@@ -3290,114 +3292,109 @@ static void BattleController_UseMove(BattleSystem *battleSys, BattleContext *bat
+            if (DEFENDER_TURN_FLAGS.enduring) {
+                battleCtx->moveStatusFlags |= MOVE_STATUS_ENDURED;
             } else {
-                param1->selfTurnFlags[param1->attacker].shellBellDamageDealt += param1->damage;
+                battleCtx->moveStatusFlags |= MOVE_STATUS_ENDURED_ITEM;
             }
+        }
 
-            param1->turnFlags[param1->defender].lastDamageTaken = param1->damage;
-            param1->turnFlags[param1->defender].lastAttacker = param1->attacker;
-            param1->msgBattlerTemp = param1->defender;
-            param1->hpCalcTemp = param1->damage;
+        battleCtx->storedDamage[battleCtx->defender] += battleCtx->damage;
 
-            BattleSystem_LoadScript(param1, 1, (0 + 2));
+        if (battleCtx->battleMons[battleCtx->defender].timesDamaged < 0xFF) {
+            battleCtx->battleMons[battleCtx->defender].timesDamaged++;
+        }
 
-            param1->command = 21;
-            param1->commandNext = 28;
-            param1->battleStatusMask |= 0x2000;
+        if (CURRENT_MOVE_DATA.class == CLASS_PHYSICAL) {
+            DEFENDER_TURN_FLAGS.physicalDamageTakenFrom[battleCtx->attacker] = battleCtx->damage;
+            DEFENDER_TURN_FLAGS.physicalDamageLastAttacker = battleCtx->attacker;
+            DEFENDER_TURN_FLAGS.physicalDamageAttackerMask |= FlagIndex(battleCtx->attacker);
+            DEFENDER_SELF_TURN_FLAGS.physicalDamageTaken = battleCtx->damage;
+            DEFENDER_SELF_TURN_FLAGS.physicalDamageLastAttacker = battleCtx->attacker;
+        } else if (CURRENT_MOVE_DATA.class == CLASS_SPECIAL) {
+            DEFENDER_TURN_FLAGS.specialDamageTakenFrom[battleCtx->attacker] = battleCtx->damage;
+            DEFENDER_TURN_FLAGS.specialDamageLastAttacker = battleCtx->attacker;
+            DEFENDER_TURN_FLAGS.specialDamageAttackerMask |= FlagIndex(battleCtx->attacker);
+            DEFENDER_SELF_TURN_FLAGS.specialDamageTaken = battleCtx->damage;
+            DEFENDER_SELF_TURN_FLAGS.specialDamageLastAttacker = battleCtx->attacker;
         }
+
+        if (DEFENDING_MON.curHP + battleCtx->damage <= 0) {
+            ATTACKER_SELF_TURN_FLAGS.shellBellDamageDealt += DEFENDING_MON.curHP * -1;
+        } else {
+            ATTACKER_SELF_TURN_FLAGS.shellBellDamageDealt += battleCtx->damage;
+        }
+
+        DEFENDER_TURN_FLAGS.lastDamageTaken = battleCtx->damage;
+        DEFENDER_TURN_FLAGS.lastAttacker = battleCtx->attacker;
+        battleCtx->msgBattlerTemp = battleCtx->defender;
+        battleCtx->hpCalcTemp = battleCtx->damage;
+
+        LOAD_SUBSEQ(BATTLE_SUBSEQ_UPDATE_HP);
+        battleCtx->command = BATTLE_CONTROL_EXEC_SCRIPT;
+        battleCtx->commandNext = BATTLE_CONTROL_AFTER_MOVE_MESSAGE;
+        battleCtx->battleStatusMask |= SYSCTL_MOVE_HIT;
     } else {
-        param1->command = 28;
+        battleCtx->command = BATTLE_CONTROL_AFTER_MOVE_MESSAGE;
     }
```

Comment:
I'm annoyed that I can't seem to move this else-branch upward and remove an indentation level from the rest of the function. Oh well.